src/containers/Login/Login.js

```diff
@@ -0,0 +1,127 @@
+import { isLoggedIn } from "../../Helpers/authHelpers";
+export class UnconnectedLogin extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {
+        email: "",
+        password: ""
+      },
+      errors: {}
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+  componentDidMount() {
+    if (isLoggedIn()) {
+      const { history, ShowMessage } = this.props;
+      ShowMessage("Hello, you are already Logged in.");
+      history.push("/");
+    }
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const { history, ShowMessage } = this.props;
+    ShowMessage("Logging in....");
+    return apiCalls("post", "/users/login", this.state)
+      .then(response => {
+        if (response.data != null) {
+          const { token, email, username } = response.data.user;
+          localStorage.setItem("token", token);
+          localStorage.setItem("email", email);
+          localStorage.setItem("username", username);
+          localStorage.setItem("user", response.data.user);
+          history.push("/");
+          ShowMessage("Log in successful");
+        } else {
+          ShowMessage({
+            message: "Opps no data returned. Try logging in again",
+            type: "error"
+          });
+        }
+      })
+      .catch(err => {
+        const { status } = err.response;
+        if (status === 500) {
+          ShowMessage({
+            message: "Api Server error 500. Try logging in again",
+            type: "error"
+          });
+        }
+        if (status === 400) {
+          const { errors } = err.response.data;
+          this.setState({ errors });
+          const { error } = this.state.errors;
+          ShowMessage({ message: error[0], type: "error" });
+        }
+      });
+  }
+  handleChange(event) {
+    const { user } = this.state;
+    this.setState({
+      user: { ...user, [event.target.name]: event.target.value },
+      errors: {}
+    });
+  }
+  render() {
+    const { email, password } = this.state.user;
+    const { errors } = this.state;
+    return (
+      <div className="row valign-wrapper login-box">
+        <div className="col s12 m4 offset-m4 valign">
+          <div className="card hoverable">
+            <form onSubmit={this.handleSubmit} data-test="comp-login">
+              <div className="card-content">
+                <span className="card-title center-align">Login</span>
+
+                <div className="row">
+                  <Input
+                    label="Email"
+                    type="email"
+                    name="email"
+                    id="email"
+                    error={errors.email ? errors.email[0] : null}
+                    className={errors.email ? "invalid" : ""}
+                    required="required"
+                    value={email}
+                    onChange={this.handleChange}
+                  />
```

Comment:
Going forward it is a good idea to have Input as an independent component so that you can reuse with different forms. You can collaborate with someone working on signup to fix this.

Reply:
Its already a reusable component 
The file `src/components/Input/Input.js`